data/src/test/java/com/linkedin/data/codec/entitystream/TestProtobufDataEncoder.java

```diff
@@ -0,0 +1,59 @@
+/*
+    Copyright (c) 2020 LinkedIn Corp.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.linkedin.data.codec.entitystream;
+
+import com.linkedin.data.ByteString;
+import com.linkedin.data.ChunkedByteStringCollector;
+import com.linkedin.data.DataComplex;
+import com.linkedin.data.DataList;
+import com.linkedin.data.DataMap;
+import com.linkedin.data.TestUtil;
+import com.linkedin.data.codec.CodecDataProviders;
+import com.linkedin.data.codec.ProtobufDataCodec;
+import com.linkedin.entitystream.CollectingReader;
+import com.linkedin.entitystream.EntityStream;
+import com.linkedin.entitystream.EntityStreams;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+
```

Comment:
What percentage of test coverage does this provide for ProtobufDataEncoder.java (in-fact for AbstractDataEncoder)  ?

Reply:
82%. missing null and IOException cases. will try to add them